### PR TITLE
Add authentication to functions runtime [AM-550]

### PR DESCRIPTION
**Pull request by mikedeploysdrones**
https://dronedeploy.atlassian.net/browse/AM-550

Added authentication to functions runtime

### CODE REVIEW

File: lib/authentication.js [hunk 1]
```diff
@@ -1,8 +1,57 @@
...
-  process.nextTick(cb(null, [
-    mockJWKS
-  ]));
-}
+const fetch = require('node-fetch');
+const jwt = require('../helpers/jwt');
+
+
+exports.getPublicKeys = function () {
+  let env = global.ENV || "test";
```
[the remark this answers is not in the document]
Comment (mikedeploysdrones) [reply lib/authentication.js]: 👌

File: modules/datastore/index.js [hunk 2]
```diff
@@ -28,58 +28,41 @@ function table(ctx) {
 }
 
 function mutateRow(ctx) {
-  return function (operation) {
+  return function (operation) {  // operation could be one of add, edit, delete, etc.
     return function (tableId, externalId, data, exclude_fields) {
       exclude_fields = exclude_fields || [];
```
Comment (miketheprogrammer) [inline modules/datastore/index.js]: nice job on cleaning up this promise. :)

**Comment by miketheprogrammer** [comment]
LGTM

**Comment by miketheprogrammer** [comment]
Reverting seems to be broken with example. need to test in morning

**Pull request merged**